wikitools/link_checker.py

```diff
@@ -68,6 +69,10 @@ def check_link(
         # this is safe to do since the caller iterates over a copy of all_articles -> we can modify it as we wish
         all_articles[raw_path] = article_parser.parse(target_file)
     target_article = all_articles[raw_path]
+
+    if target_article.front_matter.get("outdated", False) and not check_outdated:
+        return None
```

Comment:
there's a slight difference from 

> - links in outdated articles that point to changed headings in the pr won't get their fragment checked
> - links in the pr pointing to headings in outdated articles won't get their fragment checked

as we discussed, to what this pr does

> - links in *any article* pointing to headings in outdated articles won't get their fragment checked

can ignore the pr aspect of it, but this only checks if the target article is outdated